### DISCUSSION TITLE
Problems with current version

### DISCUSSION THREAD

**Discussion by jklwn4** [nidud/asmc#17] [General]
Today i downloaded and installed the actual version. In one of my projects  i get this: 

```
AND BYTE PTR [XDI], not .mask TESTTYPE._r40.B1
error A2008: syntax error : .mask TESTTYPE._r40.B1

invoke MessageBoxA, 0, offset _tstack, offset _p_TESTME, 01010h
D:\Jkb\Jk-Ide\Basic\Tests\Var\Udt_Assign.asm(3058) : error A2114: INVOKE argument type mismatch : 2

```
"addr" instead of "offset" makes the second error go away. Did i miss something? My previous version worked like charm with this code.

**Comment by nidud** [comment]
> ```
> AND BYTE PTR [XDI], not .mask TESTTYPE._r40.B1
> ```

All these dot-name directives are removed. Use [MASKOF](../blob/master/doc/operator/operator-maskof.md) or [MASK](../blob/master/doc/operator/operator-mask.md).

> ```
> invoke MessageBoxA, 0, offset _tstack, offset _p_TESTME, 01010h
> ```
Should work if the proto for argument is pointer-size and the argument is a global or local label.

**Reply by jklwn4** [reply]
MASKOF and ADDR solve the problem - thanks

...just another thing: 
would it be possible to have register names and keywords (e.g. "cx", "offset", etc.) as variable names and/or as structure members (e.g. SIZEL has a member cx, which must be renamed to "_cx" or whatever in order to make it work) ?

**Comment by nidud** [comment]
I have addressed some of the [most commonly used words](../blob/master/doc/command/option-zne.md) as directives but for reserved words that's a bit more complicated. The standard convention for this is set by [H2INC](../blob/master/source/tools/h2inc/h2inc.asm#L117), which is to prepend an underscore.

**Comment by nidud** [comment]
Note that the link option is implemented in the new version so you have to add the `-c` command line switch to avoid linking.
